### PR TITLE
Load Restart: Support Basic Quantities from Analytic Aquifers

### PR DESCRIPTION
**Pull request by bska**
This adds a basic protocol for retrieving simple scalar data for each analytic aquifer, specifically from `XAAQ`.  At the moment we only support retrieving the aquifer pressure and the total produced liquid volume from the aquifer and into the reservoir model.

### CODE REVIEW

**Comment by bska** [comment]
Marking this as ready for review and launching an initial build test.

**Comment by bska** [comment]
jenkins build this please

File: opm/output/data/Aquifer.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+  Copyright 2019 Statoil ASA.
```
Comment (GitPaean) [inline opm/output/data/Aquifer.hpp]: Equinor?

**Comment by GitPaean** [comment]
> At the moment we only support retrieving the aquifer pressure and the total produced liquid volume from the aquifer and into the reservoir model.

If I understand correctly, this statement needs to be updated? It also retrieves the initial pressure, initial volume? , or something more?

Does this PR covers writing the aquifer information to the restart files?

**Comment by bska** [comment]
> > At the moment we only support retrieving the aquifer pressure and the total produced liquid volume from the aquifer and into the reservoir model.
> 
> If I understand correctly, this statement needs to be updated? It also retrieves the initial pressure, initial volume? , or something more?

Correct.  We read the following items for all analytic aquifers

- Current aquifer pressure
- Total produced liquid volume
- Reference pressure datum depth

Then, specifically for Fetkovich aquifers, we also read the following items

- Initial aquifer volume
- Productivity index
- Time constant `Tc`.

> Does this PR covers writing the aquifer information to the restart files?

It does not.  This is read-only for now.

**Comment by joakim-hove** [comment]
>>Does this PR covers writing the aquifer information to the restart files?

>It does not. This is read-only for now.

That implies that the restart file must be *written* by another simulator? Write capabilities is in the pipeline?

File: opm/output/data/Aquifer.hpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    };
+
+    struct FetkovichData {
+        double initVolume;
+        double prodIndex;
+        double timeConstant;
+    };
+
+    struct AquiferData {
+        int aquiferID;    //< One-based ID, range 1..NANAQ
```
Comment (joakim-hove) [inline opm/output/data/Aquifer.hpp]: We probably disagree here - but to me comments stating quite obvious information like:
```
double pressure;   //< Aquifer pressure
```
is detrimental to the code quality. It is a bit like crying wolf - and the lesson I learn is to just ignore the comments.

**Comment by bska** [comment]
> > > Does this PR covers writing the aquifer information to the restart files?
> 
> > It does not. This is read-only for now.
> 
> That implies that the restart file must be _written_ by another simulator?

Correct.  I'm building this feature piecemeal.  My immediate use-case is starting Flow from a known good solution provided by another simulator.  This PR, accompanied by OPM/opm-simulators#21997, is sufficient to achieve that milestone.

> Write capabilities is in the pipeline?

Yes.  That's my next milestone.

**Comment by joakim-hove** [comment]
Looks fine to me - just merge!

**Comment by bska** [comment]
> Looks fine to me - just merge!

Very good. Merging now.  Thanks!

**Comment by GitPaean** [comment]
> It does not. This is read-only for now.

When the writing is done, we should add a restart running test for aquifer.

File: opm/output/data/Aquifer.hpp (hunk 1 above)
Comment (bska) [reply opm/output/data/Aquifer.hpp]: How did that happen?  Thanks for noticing.  I'll fix it on the next update.

**Pull request merged**